pkg/gatewayserver/gatewayserver_test.go

```diff
@@ -392,6 +405,127 @@ func TestGatewayServer(t *testing.T) {
 				}
 			},
 		},
+		{
+			Protocol:                "basicstation",
+			GatewayStatusMsgSupport: false,
+			ValidAuth: func(ctx context.Context, ids ttnpb.GatewayIdentifiers, key string) bool {
+				// TODO: Add basic Auth header check
+				return ids.EUI != nil
+			},
+			Link: func(ctx context.Context, t *testing.T, ids ttnpb.GatewayIdentifiers, key string, upCh <-chan *ttnpb.GatewayUp, downCh chan<- *ttnpb.GatewayDown) error {
+				// TODO: Add basic Auth header check
+				if ids.EUI == nil {
+					t.SkipNow()
+				}
+				wsConn, _, err := websocket.DefaultDialer.Dial("ws://0.0.0.0:1885/api/v3/gs/io/basicstation/traffic/"+registeredGatewayID, nil)
+				if !a.So(err, should.BeNil) {
+					return fmt.Errorf("Connection failed: %v", err)
+				}
+				defer wsConn.Close()
+
+				errCh := make(chan error, 2)
+				// Write upstream.
+				go func() {
+					for {
+						select {
+						case <-ctx.Done():
+							return
+						case msg := <-upCh:
+							for _, uplink := range msg.UplinkMessages {
+								var payload ttnpb.Message
+								if err := lorawan.UnmarshalMessage(uplink.RawPayload, &payload); err != nil {
+									t.Logf("Failed to unmarshal Uplink Message. Skipping...")
```

Comment:
Isn't this an error?

Reply:
I don't think that it should be. If there are 2 uplinks and one of them is garbage, the other one must be passed through (similar to the other frontends).
One thing I can do is to replace the log line with a comment to make it explicit that we are ignoring this.

Reply:
OK I see.